ChessVariantsLogic/PieceExporter.cs

```diff
@@ -0,0 +1,129 @@
+        {
+            Movement = exportPattern(piece.GetAllMovementPatterns()),
+            Captures = exportPattern(piece.GetAllCapturePatterns()),
+            Royal = piece.Royal,
+            CanBeCaptured = piece.CanBeCaptured,
+            PieceClassifier = piece.PieceClassifier,
+            PieceIdentifier = piece.PieceIdentifier,
+            Repeat = piece.Repeat,
+        };
+        return pieceState;
+    }
+
+    public static string ExportLegalMovesAsJson(HashSet<string> moveSet)
+    {
+        var moveDict = new Dictionary<string, List<string>>();
+
+        foreach (var move in moveSet)
+        {
+            var fromTo = MoveWorker.ParseMove(move);
+            if (fromTo == null)
+                throw new InvalidOperationException($"Could not parse move {move}");
+
+            var moveList = moveDict.GetValueOrDefault(fromTo.Item1, new List<string>());
+            if (moveList.Count == 0)
+            {
+                moveDict[fromTo.Item1] = moveList;
+            }
+            moveList.Add(fromTo.Item2);
+        }
+        return GameExporter.ExportMovesAsJson(moveDict);
+    }
+
+    private static List<Pattern> exportPattern(IEnumerable<IPattern> movement)
+    {
+        var patterns = new List<Pattern>();
+        foreach (var p in movement)
+        {
+            patterns.Add(new Pattern
+            {
+                XDir = p.XDir,
+                YDir = p.YDir,
+                MinLength = p.MinLength,
+                MaxLength = p.MaxLength,
+            });
+        }
+        return patterns;
+    }
+
+}
+
+/// <summary>
+/// Represents an exportation of the state of a piece and
+/// can be exported to a JSON formatted string for communication between entities.
+/// </summary>
+public record PieceState
+{
+    [JsonProperty("movements")]
+    public List<Pattern> Movement { get; set; } = null!;
+
+    [JsonProperty("captures")]
+    public List<Pattern> Captures { get; set; } = null!;
+
+    [JsonProperty("royal")]
+    public bool Royal { get; set; }
+
+    [JsonProperty("belongs_to")]
+    public PieceClassifier PieceClassifier { get; set; }
+
+    [JsonProperty("repeat")]
+    public int Repeat { get; set; }
+
+    [JsonProperty("identifier")]
+    public string PieceIdentifier { get; set; } = null!;
+
+    [JsonProperty("capturable")]
+    public bool CanBeCaptured { get; set; }
+
+    public string AsJson()
+    {
+        return JsonConvert.SerializeObject(this,
+            new JsonSerializerSettings { TypeNameHandling = TypeNameHandling.Auto });
+    }
+}
+
+/// <summary>
+/// Represents a Json-object of an <see cref="IPattern"/>.
+/// </summary>
+public record Pattern
+{
+    [JsonProperty("x_dir")]
+    public int XDir { get; set; }
+
+    [JsonProperty("y_dir")]
+    public int YDir { get; set; }
+
+    [JsonProperty("min_length")]
+    public int MinLength { get; set; }
+
+    [JsonProperty("max_length")]
+    public int MaxLength { get; set; }
+    
+}
```

Comment:
I found this out while working on the SignalR communication last week, so this is pretty hard to just now but:

Keep `JsonProperty` names the same format as properties, except that `JsonProperty` names should be camelCase.

Example:
Instead of:
```
[JsonProperty("capturable")]
public bool CanBeCaptured { get; set; }
```
Use:
```
[JsonProperty("canBeCaptured")]
public bool CanBeCaptured { get; set; }
```

**Why??**
In this case, when a client receives data from the server via SignalR they can either get
1. A json-formatted string, which they can parse into a javascript object.
2. An actual javascript object.

When we send and object immediately the keys will be formatted according to camelCase (i.e `HelloWorld` will become ``helloWorld`), in other words: the first letter is made lowercase.

In order to keep the data consistent, we should use the same keys no matter if they get an object or need to parse it into an object themselves.

Reply:
Ah I see, thanks for the tip! Should be resolved now.